src/BinSkim.Rules/PERules/BA2006.BuildWithSecureTools.cs

```diff
@@ -106,10 +109,9 @@ public override void AnalyzePortableExecutableAndPdb(BinaryAnalyzerContext conte
 
             var goodCompilers = new HashSet<string>();
 
-            var badModules = new List<ObjectModuleDetails>();
             StringToVersionMap allowedLibraries = context.Policy.GetProperty(AllowedLibraries);
 
-            var languageToBadModules = new Dictionary<Language, List<ObjectModuleDetails>>();
```

Comment:
badModules list was being updated and languageToBadModules weren't. Changed the logic to use the dictionary variable and that will update the string minimumRequiredCompilers which was always blank (review baseline)